test/errors/incompatibleKinds-detailed.good

```diff
@@ -0,0 +1,49 @@
+─── error in incompatibleKinds.chpl:3 [IncompatibleKinds] ───
+  A regular variable cannot be initialized with a type.
+  In the following initialization expression:
+      |
+    3 | var a = int;
+      |         ⎺⎺⎺  
+      |
+  the initialization expression is a type 'int(64)'.
+
+─── error in incompatibleKinds.chpl:4 [IncompatibleKinds] ───
+  A type variable cannot be initialized using a regular value.
+  In the following initialization expression:
+      |
+    4 | type b = notAType;
+      |          ⎺⎺⎺⎺⎺⎺⎺⎺  
+      |
+  the initialization expression is a value of type 'string'.
+  If you were trying to extract the type of the expression on the left of the '=', try using '.type'?
+
+─── error in incompatibleKinds.chpl:7 [IncompatibleKinds] ───
+  A 'param' cannot be initialized using a non-'param' value.
+  In the following initialization expression:
+      |
+    7 | param d = c;
+      |           ⎺  
+      |
+  the initialization expression is a value of type 'int(64)'.
+  Note that 'param' values must be known at compile-time.
+
+─── error in incompatibleKinds.chpl:10 [IncompatibleKinds] ───
+  A regular variable cannot be initialized with a type.
+  In the following initialization expression:
+       |
+    10 | var e = MyRecord;
+       |         ⎺⎺⎺⎺⎺⎺⎺⎺  
+       |
+  the initialization expression is a type 'MyRecord'.
+  If you're trying to create a new value of type 'MyRecord', try writing 'new MyRecord()' instead
+
+─── error in incompatibleKinds.chpl:13 [IncompatibleKinds] ───
+  A regular variable cannot be initialized with a type.
+  In the following initialization expression:
+       |
+    13 | var f = aliasToRecord;
+       |         ⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺  
+       |
+  the initialization expression is a type 'MyRecord'.
+  If you're trying to create a new value of type 'MyRecord', try using the 'new' keyword
```

Comment:
I think the problem here is largely technical; how do we print out the alias? (e.g., what if the user's `aliasToRecord` call is split across multiple lines?) It seemed to me like there were more opportunities for garbled output here, though I agree that printing w.r.t. the root type is unfortunate.

Reply:
I'm confused, though maybe my own feedback was confusing.  Can we not say "the initialization expression is a type 'aliasToRecord'.  If you're trying to create a new value of type 'aliasToRecord', try using the 'new' keyword.

[also, I'm noticing belatedly that there's a period missing at the end of this sentence.]

Reply:
I think that at the time we're reporting the error, the type has already been resolved to `MyRecord`, and `aliasToRecord` remains only in the syntax. If we tried to print out the exact expression used to compute the type `MyRecord`, in general, we might end up dumping a _lot_ of text, or just generally looking pretty garbled.

Reply:
Huh...  I think it'd be a good goal for dyno to preserve type aliases all the way through to code generation if at all possible.  Not doing so in the production compiler has definitely been a thorn in our side at times.  For example, for code like:

```chapel
type age = int;
var x: age;
```

it'd be far preferable to have the C code when using the C back-end be:

```chapel
typedef int age;
age x;
```

rather than:

```chapel
int x;
```

and our failure to do so has caused pain and suffering at times.  We've tried a few times to preserve type aliases in the production compiler and failed in the time allotted.  It'd be nice for dyno to succeed where we failed.

(That's not to say you need to address this in this PR of course, but maybe we could open an issue for it?  And arguably we could make this a future to show what we want to generate).

Reply:
I've [opened an issue](https://github.com/chapel-lang/chapel/issues/20914)